### ISSUE TITLE
Improve AOT code generation

### ISSUE THREAD

**Issue by johnmccutchan** [google/vector_math.dart#320]
See https://github.com/dart-lang/sdk/issues/55542

```
I would recommend that package:vector_math plasters @pragma('vm:prefer-inline') all over its APIs as a workaround.
```

**Comment by kerberjg** [comment]
Has there been any work done on this? I'm happy to help out with a PR if that's welcome 🤗

**Comment by spydon** [comment]
Nothing has been done regarding that afaik, I would say go for it! :)

**Comment by spydon** [comment]
Still up for doing this @kerberjg? If so I'll assign you to the issue.

**Comment by spydon** [comment]
@kerberjg read [this](https://github.com/google/vector_math.dart/pull/347) and [this](https://github.com/google/vector_math.dart/pull/345/) first, it is not trivial to know where they can be added.

**Comment by kerberjg** [comment]
@spydon yup I saw these, I'm gonna do a before/after benchmark and leave a note wherever an inline results in a perf decrease

Go ahead and assign me please 🥳
